mne/viz/utils.py

```diff
@@ -167,43 +167,57 @@ def mne_analyze_colormap(limits=[5, 10, 15], format='mayavi'):
         brain.scale_data_colormap(fmin=-15, fmid=0, fmax=15, transparent=False)
 
     """
-    l = np.asarray(limits, dtype='float')
-    if len(l) != 3:
-        raise ValueError('limits must have 3 elements')
-    if any(l < 0):
-        raise ValueError('limits must all be positive')
-    if any(np.diff(l) <= 0):
+    # Ensure limits is an array
+    limits = np.asarray(limits, dtype='float')
+
+    if len(limits) != 3 and len(limits) != 6:
```

Comment:
This is a feature @dengemann wanted. It allows for custom mne-analyze colormaps that have nonsymmetric control points. We figured it was too complex to put in the standard plotting functions.